include/swift/SIL/SILFunction.h

```diff
@@ -250,6 +255,9 @@ class SILFunction
   llvm::SmallVector<SILReverseDifferentiableAttr *, 4>
     ReverseDifferentiableAttrs;
 
+  /// The function's `[compiler_evaluable]` attribute.
+  SILCompilerEvaluableAttr *CompilerEvaluableAttr = nullptr;
```

Comment:
Magical behavior can be encoded in the `@_semantics` attribute, no? Having a tail-allocated field is a bit heavy weight. IMO this should just be a boolean field.

Reply:
I agree with richard in the short term, but in the long term it makes sense to define a principled attribute of some kind.  In the short term, this will be a private implementation detail of the stdlib, so using @_semantics is "cheap" in terms of implementation complexity, so I'd recommend going with it.

Reply:
Oh wait, I misunderstood which attribute this is.  This is the actual @compilerEvaluable attribute you want the checker to pay attention to.  This really does need an explicit representation of some type: either an attribute or a bool in SILFunction.